src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,153 @@
+package duke.task;
+
+import duke.DukeException;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Manages a list of tasks.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
+
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Adds a task to the list. Throws DukeException if the description is invalid.
+     * @param taskType taskType.
+     * @param description description.
+     * @return success message and the task added.
+     * @throws DukeException if the description is invalid.
+     */
+    public String addTask(TaskType taskType, String description) throws DukeException {
+        description = description.trim();
+        switch (taskType) {
+        case TODO:
+            if (!description.isEmpty()) {
+                this.tasks.add(new ToDo(description));
+            } else {
+                throw new DukeException("todo error");
+            }
+            break;
+        case DEADLINE:
+            Pattern p = Pattern.compile("(.+) /by (.+)");
+            Matcher m = p.matcher(description);
+            if (m.matches() && !m.group(1).isEmpty() && !m.group(2).isEmpty()) {
+                try {
+                    this.tasks.add(new Deadline(m.group(1), m.group(2)));
+                } catch (DateTimeParseException e) {
+                    throw new DukeException("deadline error");
+                }
+            } else {
+                throw new DukeException("deadline error");
+            }
+            break;
+        default:
+            Pattern pattern = Pattern.compile("(.+) /from (.+) /to (.+)");
+            Matcher matcher = pattern.matcher(description);
+            if (matcher.matches() && !matcher.group(1).isEmpty() && !matcher.group(2).isEmpty()
+                    && !matcher.group(3).isEmpty()) {
+                this.tasks.add(new Event(matcher.group(1), matcher.group(2), matcher.group(3)));
+            } else {
+                throw new DukeException("event error");
+            }
+            break;
+        }
```

Comment:
Do you think it will be better to not put event as your default case? In case there is a need to add extra features with different commands

Reply:
Good suggestion. Thanks!!